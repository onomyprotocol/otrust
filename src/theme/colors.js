export const light = {
  colors: {
    primary: '#4877c1',
    primaryHover: '#335894',
    secondary: '#8991a1',
    surface: '#fafafc',
    textBody: '#36364b',
    textPrimary: '#4877C1',
    white: '#ffffff',
    light: '#dfe4ed',
    icon: '#a7b0c3',
    bgLight: '#f0f2f5',
    placeholder: '#9e9e9f',
    red: '#e86250',
    placeholderLight: '#aab0c3',
  },
};

export const dark = {
  colors: {
    primary: '#4877c1',
    primaryHover: '#335894',
    secondary: '#8991a1',
    surface: '#fafafc',
    textBody: '#36364b',
    textPrimary: '#4877C1',
    white: '#ffffff',
    light: '#dfe4ed',
    icon: '#a7b0c3',
    bgLight: '#f0f2f5',
    placeholder: '#9e9e9f',
    red: '#e86250',
    placeholderLight: '#aab0c3',
  },
};

// I added new theme for incremental update, when the CSS work is done, old themes could be deleted and "New" part from the name could be removed via replace all
export const darkNew = {
  colors: {
    highlightBlue: '#88D1FF',
    highlightYellow: '#FFDDA1',
    highlightGreen: '#7CF9BA',
    highlightRed: '#C75A5A',
    textPrimary: '#E1DFEB',
    textSecondary: '#9895A6',
    textThirdly: '#656273',
    iconsNormal: '#8D89A3',
    bgHighlightBorder: '#302E3D',
    bgNormal: '#1F1D29',
    bgDarken: '#1A1723',
    bgDarkest: '#06060f',
  },
};
