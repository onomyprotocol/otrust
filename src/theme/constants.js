export const responsive = {
  laptop: '1366px',
  laptopSmall: '1180px',
  tablet: '1024px',
  tabletSmall: '834px',
  smartphoneLarge: '700px',
  smartphone: '480px',
  smartphoneSmall: '375px',
};
