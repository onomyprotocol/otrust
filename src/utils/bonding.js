const a = 100000000;

export function priceAtSupply(supply) {
  return (supply / a) ** 2;
}

// Bonding curve price at specified supply
export function supplyAtPrice(price) {
  //#NOM Sold = sqrt(ETH/NOM) * a
  return Math.sqrt(price) * a;
}

// NOM supply range to ETH
export function NOMsupplyETH(supplyTop, supplyBot) {
  // Integrate over curve to get amount of ETH needed to buy amount of NOM
  // ETH = a/3((supplyNOM_Top/a)^3 - (supplyNOM_Bot/a)^3)
  return (a / 3) * ((supplyTop / a) ** 3 - (supplyBot / a) ** 3);
}

// Returns Buy Quote for the purchase of NOM based on amount of ETH
// Parameters:
// Input
// amount: amount of ETH
// Output
// output: amount of NOM
export function ETHtoNOM(amount, supply) {
  const price = priceAtSupply(supply);
  // 1. Determine supply bottom
  // Price bottom is 1% above priceBondCurve
  const priceBot = price * 1.01;
  const supplyBot = supplyAtPrice(priceBot);
  // 2. Integrate over curve, and solve for supply top
  // (3*ETH/a + (supplyNOM_Bot/a)^3)^(1/3)
  const supplyTop = a * ((3 * amount) / a + supplyBot ** 3 / a ** 3) ** (1 / 3);
  // 3. Subtract supply bottom from top to get amount NOM for amount ETH
  const diff = supplyTop - supplyBot;
  return { supplyBot, supplyTop, diff };
}

// Returns Sell Quote: NOM for ETH (Dec 18)
// Parameters:
// Input
// uint256 amountNOM: amount of NOM to be sold (18 decimal)
// Output
// uint256: amount of ETH paid in Wei or ETH (18 decimal)
export function NOMtoETH(amount, supplyTop) {
  const supplyBot = supplyTop - amount;
  const diff = 0.99 * NOMsupplyETH(supplyTop, supplyBot);
  return { supplyBot, supplyTop, diff };
}
