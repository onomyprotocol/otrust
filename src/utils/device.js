const size = {
  mobileS: '320px',
  mobileM: '375px',
  mobileL: '425px',
  tablet: '768px',
  laptop: '1024px',
  desktopS: '1366px',
  desktopM: '1920px',
  desktopL: '2560px',
};

export const device = {
  mobileS: `(min-width: ${size.mobileS})`,
  mobileM: `(min-width: ${size.mobileM})`,
  mobileL: `(min-width: ${size.mobileL})`,
  tablet: `(min-width: ${size.tablet})`,
  laptop: `(min-width: ${size.laptop})`,
  desktopS: `(min-width: ${size.desktopS})`,
  desktopM: `(min-width: ${size.desktopM})`,
  desktopL: `(min-width: ${size.desktopL})`,
};
