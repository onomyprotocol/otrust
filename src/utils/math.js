import { BigNumber } from 'bignumber.js';

export function chopFloat(value, precision) {
  return Math.round(value * 10 ** precision) / 10 ** precision;
}

export function randomInt(min, max) {
  return Math.floor(Math.random() * (max - min + 1) + min);
}

export function truncate(str, maxDecimalDigits) {
  if (str.includes('.')) {
    const parts = str.split('.');
    return parts[0] + '.' + parts[1].slice(0, maxDecimalDigits);
  }
  return str;
}

export function format18(bignumber) {
  return bignumber.div(new BigNumber(10 ** 18));
}

export function parse18(bignumber) {
  return bignumber.times(new BigNumber(10 ** 18));
}

export var isNumber = function isNumber(value) {
  return typeof value === 'number' && isFinite(value);
};
