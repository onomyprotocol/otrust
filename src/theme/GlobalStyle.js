import { createGlobalStyle } from 'styled-components';

import BebasNeueBold from 'assets/fonts/BebasNeueBold.ttf';
import BebasNeueBook from 'assets/fonts/BebasNeueBook.ttf';
import BebasNeueRegular from 'assets/fonts/BebasNeueRegular.ttf';
import { responsive } from './constants';

export const GlobalStyle = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }

  :root {
    position: relative;

    background: linear-gradient(135deg, #16161f, #06060f);

    color: ${props => props.theme.colors.textPrimary};
    font-size: 14px;
    font-family: 'Poppins', sans-serif;
  
    @media screen and (max-width: ${responsive.laptop}) {
      font-size: 12px;
    }
  }
  
  html, body {
    min-height: 100%;
  }

  button {
    cursor: pointer;

    &:focus {
      outline: none;
    }
  }

  @font-face {
    font-family: 'Bebas Neue';
    font-weight: 300;
    src: url('${BebasNeueBook}') format("truetype");
  }

  @font-face {
    font-family: 'Bebas Neue';
    font-weight: 400;
    src: url('${BebasNeueRegular}') format('truetype');
  }
  
  @font-face {
    font-family: 'Bebas Neue';
    font-weight: 600;
    src: url('${BebasNeueBold}') format("truetype");
  }

  body {
    font-family: 'Poppins', sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }
`;
