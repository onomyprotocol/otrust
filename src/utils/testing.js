import { ThemeProvider } from 'styled-components';
import { render } from '@testing-library/react';

import { dark } from 'Theme/theme';

export const renderWithTheme = (Component, rest) => {
  return render(
    <ThemeProvider theme={dark}>
      <Component {...rest} />
    </ThemeProvider>,
  );
};
