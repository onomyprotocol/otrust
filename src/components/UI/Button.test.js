import { cleanup, fireEvent, render } from '@testing-library/react';
import { AccentButton, Button, InvisibleButton } from './Button';
import { renderWithTheme } from '../../utils/testing';
import { ThemeProvider } from 'styled-components';
import { darkNew } from 'Theme/theme';

const testProps = {
  width: 50,
};

const clickFN = jest.fn();

describe('Given the AccentButton component and width is provided in the props', () => {
  describe('when the component is rendered', () => {
    afterEach(cleanup);

    it('should match the snapshot', () => {
      const { asFragment } = renderWithTheme(AccentButton, testProps);
      expect(asFragment()).toMatchSnapshot();
    });

    it('should work click AccentButton', () => {
      const { getByRole } = render(
        <ThemeProvider theme={darkNew}>
          <AccentButton onClick={clickFN} />
        </ThemeProvider>,
      );

      const button = getByRole('button');
      fireEvent.click(button);

      expect(clickFN).toHaveBeenCalled();
    });
  });
});

describe('Given the Button component and width is provided in the props', () => {
  describe('when the component is rendered', () => {
    afterEach(cleanup);

    it('should match the snapshot', () => {
      const { asFragment } = renderWithTheme(Button, testProps);
      expect(asFragment()).toMatchSnapshot();
    });

    it('should work click Button', () => {
      const { getByRole } = render(
        <ThemeProvider theme={darkNew}>
          <Button onClick={clickFN} />
        </ThemeProvider>,
      );

      const button = getByRole('button');
      fireEvent.click(button);

      expect(clickFN).toHaveBeenCalled();
    });
  });
});

describe('Given the Button component and width is not provided in the props', () => {
  describe('when the component is rendered', () => {
    afterEach(cleanup);

    it('should match the snapshot', () => {
      const { asFragment } = renderWithTheme(Button);
      expect(asFragment()).toMatchSnapshot();
    });

    it('should work click Button', () => {
      const { getByRole } = render(
        <ThemeProvider theme={darkNew}>
          <Button onClick={clickFN} />
        </ThemeProvider>,
      );

      const button = getByRole('button');
      fireEvent.click(button);

      expect(clickFN).toHaveBeenCalled();
    });
  });
});

describe('Given the InvisibleButton component', () => {
  describe('when the component is rendered', () => {
    afterEach(cleanup);

    it('should match the snapshot', () => {
      const { asFragment } = renderWithTheme(InvisibleButton);
      expect(asFragment()).toMatchSnapshot();
    });

    it('should work click InvisibleButton', () => {
      const { getByRole } = render(
        <ThemeProvider theme={darkNew}>
          <InvisibleButton onClick={clickFN} />
        </ThemeProvider>,
      );

      const button = getByRole('button');
      fireEvent.click(button);

      expect(clickFN).toHaveBeenCalled();
    });
  });
});
