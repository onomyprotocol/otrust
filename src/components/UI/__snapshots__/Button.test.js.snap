// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the AccentButton component and width is provided in the props when the component is rendered should match the snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-gtsrHT kpLgTl"
    width="50"
  />
</DocumentFragment>
`;

exports[`Given the Button component and width is not provided in the props when the component is rendered should match the snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-bdnxRM gLKfsM"
  />
</DocumentFragment>
`;

exports[`Given the Button component and width is provided in the props when the component is rendered should match the snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-bdnxRM fPQjMu"
    width="50"
  />
</DocumentFragment>
`;

exports[`Given the InvisibleButton component when the component is rendered should match the snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-hKFxyN kTRXmr"
  />
</DocumentFragment>
`;
